delay-io.js

```diff
@@ -138,7 +138,7 @@ var DelayedSocketBase = {
 	__run: function(fnName, args) {
 		var realSocket = this.fifoSocket.realSocket;
 		if (realSocket && realSocket[fnName]) {
-			return realSocket[fnName].apply(this, args);
+			return realSocket[fnName]();
```

Comment:
do we not need to pass the arguments?

maybe smth like this would work (pass `realSocket` as context instead of `this` to keep the method call correct):
```
return realSocket[fnName].apply(realSocket, args);
```
?